doc/developers/contributing.rst

```diff
@@ -957,49 +957,71 @@ adheres to the scikit-learn interface and standards by running
   >>> check_estimator(LinearSVC)  # passes
 
 The main motivation to make a class compatible to the scikit-learn estimator
-interface might be that you want to use it together with model assessment and
-selection tools such as :class:`model_selection.GridSearchCV`.
-
-For this to work, you need to implement the following interface.
-If a dependency on scikit-learn is okay for your code,
-you can prevent a lot of boilerplate code
-by deriving a class from ``BaseEstimator``
-and optionally the mixin classes in ``sklearn.base``.
-E.g., below is a custom classifier. For more information on this example, see
-`scikit-learn-contrib <https://github.com/scikit-learn-contrib/project-template/blob/master/skltemplate/template.py>`_::
-
-  >>> import numpy as np
-  >>> from sklearn.base import BaseEstimator, ClassifierMixin
-  >>> from sklearn.utils.validation import check_X_y, check_array, check_is_fitted
-  >>> from sklearn.utils.multiclass import unique_labels
-  >>> from sklearn.metrics import euclidean_distances
-  >>> class TemplateClassifier(BaseEstimator, ClassifierMixin):
-  ...
-  ...     def __init__(self, demo_param='demo'):
-  ...         self.demo_param = demo_param
-  ...
-  ...     def fit(self, X, y):
-  ...
-  ...         # Check that X and y have correct shape
-  ...         X, y = check_X_y(X, y)
-  ...         # Store the classes seen during fit
-  ...         self.classes_ = unique_labels(y)
-  ...
-  ...         self.X_ = X
-  ...         self.y_ = y
-  ...         # Return the classifier
-  ...         return self
-  ...
-  ...     def predict(self, X):
-  ...
-  ...         # Check is fit had been called
-  ...         check_is_fitted(self, ['X_', 'y_'])
-  ...
-  ...         # Input validation
-  ...         X = check_array(X)
-  ...
-  ...         closest = np.argmin(euclidean_distances(X, self.X_), axis=1)
-  ...         return self.y_[closest]
+interface might be that you want to use it together with model evaluation and
+selection tools such as :class:`model_selection.GridSearchCV` and
+:class:`pipeline.Pipeline`.
+
+For this to work, you need to implement the interface described below.
```

Comment:
Not sure what "described below" actually refers to, can we add a link? I know it was like this before but adding topics as you did makes below even belower if that makes sense ...